docs/Blocks.md

```diff
@@ -1,78 +1,27 @@
+The role path of an element can be constructed from its role and the roles of all its parents. The role path can then be used to target a particular section of the control model tree or to retrieve the object ids again after a restart.
 
-Where the following data types are defined:
+## Ports and signal paths
 
-```typescript
-typedef NcString NcName;
+Blocks MAY also model how nested member classes are interconnected using [ports](Framework.md#ncport) and [signal paths](Framework.md#ncsignalpath).
 
-typedef sequence<NcString>  NcRolePath;
-
-enum NcIoDirection { // Input and/or output
-    "Undefined", // 0 Flow direction is not defined
-    "Input", // 1 Samples flow into owning object
-    "Output", // 2 Samples flow out of owning object
-    "Bidirectional" // 3 For possible future use
-};
-
-interface NcPort {
-    attribute NcString      role;       // Unique within owning object
-    attribute NcIoDirection direction;  // Input (sink) or output (source) port
-    attribute NcRolePath?   clockPath;  // Rolepath of this port's sample clock or null if none
-};
-
-interface NcSignalPath {
-    attribute   NcString        role;   // Unique identifier of this signal path in this block
-    attribute   NcString?       label;  // Optional label
-    attribute   NcPortReference source;
-    attribute   NcPortReference sink;
-};
-```
-
-More information on ports and signal paths can be found in [MS-05-01](https://specs.amwa.tv/ms-05-01) under the `Device Model\Signal paths` section.
-
-Blockspecs are defined in [MS-05-03 NMOS Control Block Specifications](https://specs.amwa.tv/ms-05-03).
+More information on ports and signal paths can be found in [MS-05-01 Signal paths](https://specs.amwa.tv/ms-05-01/branches/v1.0-dev/docs/Device_Model.html#signal-paths).
 
 ## Tree discovery
 
-Blocks enable device tree discovery by offering the descriptors of their contained members in the `members` property.
-
-```typescript
-interface NcBlockMemberDescriptor {
-    attribute NcString role;                                // Role of member in its containing block
-    attribute NcOid oid;                                    // OID of member
-    attribute NcBoolean constantOid                         // TRUE iff member's OID is hardwired into device 
-    attribute NcClassId classId;                            // Class ID
-    attribute NcString? userLabel;                          // User label
-    attribute NcOid owner;                                  // Containing block's OID
-    attribute sequence<NcPropertyConstraints>? constraints  // Constraints on this member or, for a block, its members.
-};
-```
+Blocks enable device tree discovery by offering the descriptors of their contained members in the `members` property which holds a collection of type [NcBlockMemberDescriptor](Framework.md#ncblockmemberdescriptor).
```

Comment:
Perhaps normative language is needed here:

```suggestion
Blocks MUST enable device tree discovery by offering the descriptors of their contained members in the `members` property which holds a collection of type [NcBlockMemberDescriptor](Framework.md#ncblockmemberdescriptor).
```

Reply:
Yes, fair point.

On a separate point with reference to #42 which term is going to be the preferred: Device Model or Device Tree?  Given that Device Model is the term defined in [MS-05-01](https://github.com/AMWA-TV/ms-05-01/blob/e9f03c860030452ab08bc0123f240b66c1fb6ada/docs/Device%20Model.md#device-model) should we not be using that instead?